numpy/core/src/umath/umathmodule.c

```diff
@@ -58,19 +58,6 @@ object_ufunc_type_resolver(PyUFuncObject *ufunc,
     return 0;
 }
 
-static int
-object_ufunc_loop_selector(PyUFuncObject *ufunc,
-                            PyArray_Descr **NPY_UNUSED(dtypes),
-                            PyUFuncGenericFunction *out_innerloop,
-                            void **out_innerloopdata,
-                            int *out_needs_api)
-{
-    *out_innerloop = ufunc->functions[0];
-    *out_innerloopdata = (ufunc->data == NULL) ? NULL : ufunc->data[0];
-    *out_needs_api = 1;
-
-    return 0;
-}
```

Comment:
The default one works fine here.  My guess is there was some historic reasons (maybe the needs api early on) of why it was explicit.  Or just a micro-optimization, but if there is one loop to check that seems hardly necessary.